BTDBTest/ObjectDbTableTest.cs

```diff
@@ -3119,6 +3119,68 @@ public void UpsertRunningAllOnSerializeMethods()
             table.Upsert(new() { Id = 1000, Text = "ahoj" });
             Assert.Equal("AHOJ", table.First().Text);
         }
+
+        public class ItemWithLazyDictionary
+        {
+            [PrimaryKey(1)]
+            public ulong Id { get; set; }
+            public IDictionary<ulong, InnerObject> LazyDictionary { get; set; }
+
+            public class InnerObject
+            {
+                public bool Flag { get; set; }
+            }
+        }
+
+        public interface IItemWithLazyDictionaryTable : IRelation<ItemWithLazyDictionary>
+        {
+            ItemWithLazyDictionary FindById(ulong id);
+            void Update(ItemWithLazyDictionary itemWithLazyDictionary);
+        }
+
+        [Fact]
+        public void LazyDictionaryShouldStoreChanges()
+        {
+            using (var tr = _db.StartTransaction())
+            {
+                var table = tr.GetRelation<IItemWithLazyDictionaryTable>();
+                table.Upsert(new()
+                {
+                    Id = 1,
+                    LazyDictionary = new Dictionary<ulong, ItemWithLazyDictionary.InnerObject>
+                    {
+                        {1, new ItemWithLazyDictionary.InnerObject()}
+                    }
+                });
+
+                tr.Commit();
+            }
+
+            using (var tr = _db.StartTransaction())
+            {
+                var table = tr.GetRelation<IItemWithLazyDictionaryTable>();
+                foreach (var itemWithLazyDictionary in table.ToList())
+                {
+                    Assert.False(itemWithLazyDictionary.LazyDictionary[1].Flag);
+                    itemWithLazyDictionary.LazyDictionary[1].Flag = true;
```

Comment:
This calls just indexed getter so you are changing Flag in temporary object.

Correct code would be:
```
var inner = itemWithLazyDictionary.LazyDictionary[1];
inner.Flag = true;
itemWithLazyDictionary.LazyDictionary[1] = inner; // This line does actual write to DB.
```

Reply:
Another faster option in this case InnerObject does not need to be read from DB at all:
```
itemWithLazyDictionary.LazyDictionary[1] = new () { Flag = true };
```